javav2/example_code/s3/src/main/java/com/example/s3/express/S3DirectoriesActions.java

```diff
@@ -0,0 +1,551 @@
+                    }
+                    throw new CompletionException("Failed to copy object", exception);
+                }
+            });
+    }
+
+    /**
+     * Creates an asynchronous session for the specified S3 bucket.
+     *
+     * @param s3Client the S3 asynchronous client to use for creating the session
+     * @param bucketName the name of the S3 bucket for which to create the session
+     * @return a {@link CompletableFuture} that completes when the session is created, or throws a {@link CompletionException} if an error occurs
+     */
+    public CompletableFuture<Void> createSessionAsync(S3AsyncClient s3Client, String bucketName) {
+        CreateSessionRequest request = CreateSessionRequest.builder()
+            .bucket(bucketName)
+            .build();
+
+        return s3Client.createSession(request)
+            .thenRun(() -> logger.info("Created session for bucket: " + bucketName))
+            .whenComplete((ignored, exception) -> {
+                if (exception != null) {
+                    Throwable cause = exception.getCause();
+                    if (cause instanceof S3Exception) {
+                        throw new CompletionException("Couldn't create the session. Reason: " + ((S3Exception) cause).awsErrorDetails().errorMessage(), cause);
+                    }
+                    throw new CompletionException("Unexpected error occurred while creating session", exception);
+                }
+            });
+    }
+
+    /**
+     * Creates a new S3 directory bucket in a specified Zone (For example, a
+     * specified Availability Zone in this code example).
+     *
+     * @param s3Client   The S3 client used to create the bucket
+     * @param bucketName The name of the bucket to be created
+     * @param zone       The region where the bucket will be created
+     * @throws S3Exception if there's an error creating the bucket
+     */
+    public CompletableFuture<Void> createDirectoryBucketAsync(S3AsyncClient s3Client, String bucketName, String zone) {
+        logger.info("Creating bucket: " + bucketName);
+
+        CreateBucketConfiguration bucketConfiguration = CreateBucketConfiguration.builder()
+            .location(LocationInfo.builder()
+                .type(LocationType.AVAILABILITY_ZONE)
+                .name(zone)
+                .build())
+            .bucket(BucketInfo.builder()
+                .type(BucketType.DIRECTORY)
+                .dataRedundancy(DataRedundancy.SINGLE_AVAILABILITY_ZONE)
+                .build())
+            .build();
+
+        CreateBucketRequest bucketRequest = CreateBucketRequest.builder()
+            .bucket(bucketName)
+            .createBucketConfiguration(bucketConfiguration)
+            .build();
+
+        return s3Client.createBucket(bucketRequest)
+            .thenAccept(response -> logger.info("Bucket created successfully with location: " + response.location()))
```

Comment:
Move logging to the `whenComplete` method.

Reply:
Fixed